### ISSUE TITLE
revert MSTK?  

### ISSUE THREAD

**Issue by ecoon** [amanzi/amanzi#516]
There are several open issues in MSTK 3.3 suggesting odd behavior or outright seg faults.  See:

https://github.com/MeshToolkit/MSTK/issues/116

and

https://github.com/MeshToolkit/MSTK/issues/112

The second is a nuisance and a performance issue because it results in non-contiguous gids which requires a copy for every Hypre call.  The first is blocking.  @jd-moulton do you have any big meshes you can test on?

If someone else can confirm this issue, I'd like to revert MSTK.  I know 3.1 works, but will build versions of Amanzi with 3.3.1 and 3.2.1, which TPLVersions suggests we have somewhere, and test those to find the latest working version.

**Comment by jd-moulton** [comment]
Let me know which version to jump back to (newest possible), and I'll prioritize pushing an update.

**Comment by ecoon** [comment]
I'm testing 3.3.1 now -- it appears to be fine.  I was able to partition on 256 cores at least (which fit in backfill) but let me test on 320 to be sure (which is what was failing).

**Comment by ecoon** [comment]
So it ended up failing on 320.  Which is information.  I'm not sure what would cause failure on 320 but not on 256.  I've been able to bracket the regression --  partitioning works on MSTK 3.1.3, but not in 3.2.0.  I'll try and dig in on finding what is failing today.

**Comment by ecoon** [comment]
I think this is an MPI bug.

**Comment by ecoon** [comment]
I guess I'm going to close this... I spent the day with some ugly hacked debugging trying to understand why the specific messages were failing while they were part-sent, and am simply convinced at this point there are some issues in this specific version of openucx, which is used under openmpi on this machine.  Googling suggests others have had issues with this specific version, but also that it got fixed, so I'm starting to think I was just unlucky in the version of openmpi I picked.

Furthermore, a rebuild on old gcc and openmpi worked.  So at any rate I don't think it is MSTK's fault (though the other bug is still an issue in MSTK).
